### PR TITLE
Support for MacOS with the -c arg for .PKG files in FastLane

### LINKED ISSUE #116: Submit MacOS Apps .pkg file submission & not .ipa
Can this extension be modified to support MacOS application submissions please. 💻 

As when creating a package/submission for MacOS as opposed to iPhone/iPad etc you need to upload a .pkg file as opposed to an .ipa 📱

Looking at the source code you set the `-i` param for the FastLane tool
https://github.com/Microsoft/app-store-vsts-extension/blob/master/Tasks/app-store-release/app-store-release.ts#L212

Which means even if I update this tasks `ipaPath` property to be a path to a .app file instead it will not work.

Looking at the source code of FastLane from the link/mention in your own sourcecode it seems FastLane will complain that the file is not an `.ipa` & this gives me no way to upload/submit a .pkg for MacOS review.

Can I suggest that this task supports more options from FastLane to allow not just .ipa submissions

Looks like the setting I would need would be `-c`
https://github.com/fastlane/fastlane/blob/master/deliver/lib/deliver/options.rb#L53-L68

### PR DESCRIPTION
**Pull request by warrenbuckley**
Fixes #116 

* Renames ipaPath input to be appPath
* Update ipaPath description to mention it can be PKG
* Adds new dropdown for application type
* Switch logic for apptype to set the -i or -c arg
* Switch logic sets apptype with the -j arg

### CODE REVIEW

File: Tasks/app-store-release/app-store-release.ts [hunk 1]
```diff
@@ -81,7 +81,7 @@ async function run() {
             }
         }
 
-        let ipaPath: string = tl.getInput('ipaPath', true);
```
Comment (madhurig) [inline Tasks/app-store-release/app-store-release.ts]: Changing input name is a breaking change, so please leave the input name as ipaPath. Changing the variable to filePath is fine.

File: Tasks/app-store-release/task.json [hunk 2]
```diff
@@ -107,12 +107,24 @@
             "helpMarkDown": "The unique app identifier (e.g. com.myapp.etc). Required if the track is Production."
         },
         {
-            "name": "ipaPath",
+            "name": "appType",
+            "type": "pickList",
+            "label": "Application Type",
+            "helpMarkDown": "The type of application you wish to submit",
+            "required": true,
```
Comment (madhurig) [inline Tasks/app-store-release/task.json]: Add `"defaultValue": "iOS"`

File: Tasks/app-store-release/task.json [hunk 3]
```diff
@@ -107,12 +107,24 @@
...
+            "options": {
+                "iOS": "iOS",
+                "AppleTVOS": "AppleTVOS",
+                "OSX": "OSX"
+            }
+        },
+        {
+            "name": "appPath",
```
Comment (madhurig) [inline Tasks/app-store-release/task.json]: keep this as `ipaPath` for back compat.

File: Tasks/app-store-release/task.json [hunk 4]
```diff
@@ -107,12 +107,24 @@
...
         },
         {
-            "name": "ipaPath",
+            "name": "appType",
+            "type": "pickList",
+            "label": "Application Type",
+            "helpMarkDown": "The type of application you wish to submit",
+            "required": true,
+            "options": {
+                "iOS": "iOS",
```
Comment (madhurig) [inline Tasks/app-store-release/task.json]: Please change these to `iOS`, `tvOS` and `macOS` to match how they appear in Xcode and the Apple developer portal.

File: Tasks/app-store-release/task.json [hunk 5]
```diff
@@ -318,6 +330,7 @@
         "MultipleIpaFilesFound": "More than one IPA file found using pattern: %s",
         "FastlaneSessionEmpty": "'Fastlane Session' is not set in the service connection configured for two-step verification.",
         "ReleaseNotesRequiredForExternalTesting": "'What to Test?' is required when using 'Distribute to External Testers'.",
-        "ExternalTestersCannotSkipWarning": "'Skip Build Processing Wait' and 'Skip Submission' is not supported with 'Distribute to External Testers'. Please check your build configuration."
+        "ExternalTestersCannotSkipWarning": "'Skip Build Processing Wait' and 'Skip Submission' is not supported with 'Distribute to External Testers'. Please check your build configuration.",
+        "NotValidAppType": "An incorrect ApplicationType was chosen. Valid values osx, ios or appletvos. Value set: %s"
```
Comment (madhurig) [inline Tasks/app-store-release/task.json]: `iOS, macOS or tvOS`

File: Tasks/app-store-release/task.json [hunk 6]
```diff
@@ -107,12 +107,24 @@
             "helpMarkDown": "The unique app identifier (e.g. com.myapp.etc). Required if the track is Production."
         },
```
Comment (madhurig) [inline Tasks/app-store-release/task.json]: Please update minor version of the task to 150

**Comment by warrenbuckley** [comment]
Thanks for the guidance @madhurig 
I have finished work for the week. I will try and see if I find time over the weekend to make the adjustments

**Comment by warrenbuckley** [comment]
Hi @madhurig 
I think I have done all the changes you have suggested. Let me know if I have missed anything.

Thanks,
Warren :)

File: Tasks/app-store-release/app-store-release.ts [hunk 7]
```diff
@@ -128,7 +130,7 @@ async function run() {
         process.env['PATH'] = process.env['PATH'] + ':' + gemCache + path.sep + 'bin';
 
         // Ensure there's exactly one ipa before installing fastlane tools
```
Comment (joshmgross) [inline Tasks/app-store-release/app-store-release.ts]: Can we update this comment?

File: Tasks/app-store-release/app-store-release.ts [hunk 8]
```diff
@@ -209,8 +211,33 @@ async function run() {
...
             // See https://github.com/fastlane/fastlane/blob/master/deliver/lib/deliver/options.rb for more information on these arguments
             let deliverCommand: ToolRunner = tl.tool('fastlane');
-            deliverCommand.arg(['deliver', '--force', '-u', credentials.username, '-a', bundleIdentifier, '-i', ipaPath]);
+            deliverCommand.arg(['deliver', '--force', '-u', credentials.username, '-a', bundleIdentifier]);
             deliverCommand.argIf(skipBinaryUpload, ['--skip_binary_upload', 'true']);
+
+            //Sets -i or -c depending if app submission is for (-i) iOS/tvOS or (-c) MacOS
+            switch (applicationType.toLocaleLowerCase()) {
+                case 'macos':
+                    // Use the -C flag &
```
Comment (joshmgross) [inline Tasks/app-store-release/app-store-release.ts]: Is this comment complete? "& what?"

File: Tasks/app-store-release/app-store-release.ts (hunk 7 above)
Comment (madhurig) [reply Tasks/app-store-release/app-store-release.ts]: I will update the comments after merging.

**Pull request merged**